File: Dinner/Assets/NPCSelector.cs
using UnityEngine;
using System.Collections;

public class NPCSelector : MonoBehaviour {


	// Use this for initialization
	void Start () {
	
	}
	
	// Update is called once per frame
	void Update () {
		RaycastHit hit;
		if (Physics.Raycast(transform.position, transform.forward, out hit, 100)){
			if (hit.collider.tag == "AI"){
				print ("Trigger something here");
			}
		}
	
	}
}

File: Dinner/Assets/Scripts/NPCSelector.js
#pragma strict

var myself : NodListener;

function Start () {
	myself = GetComponent(NodListener);
}

function Update () {
	var hit : RaycastHit;
	if (Physics.Raycast(transform.position, transform.forward, hit)){
		if (hit.collider.gameObject.tag == "AI"){
			
			var na : NodActivatee = hit.collider.gameObject.GetComponent(NodActivatee);
			if (na.disabled){
				
				return;
			}
			if(myself.nodding){
				var cc : MyCharacterController = na.myParent;
				cc.Trigger();
			}
		} 
	}
}

